neofs/neofs_contract.go

```diff
@@ -172,16 +187,11 @@ func InnerRingCandidateRemove(key interop.PublicKey) {
 		}
 	}
 
-	nodes := []common.IRNode{} // it is explicit declaration of empty slice, not nil
-	candidates := getNodes(ctx, candidatesKey)
-
-	for i := range candidates {
-		c := candidates[i]
-		if !common.BytesEqual(c.PublicKey, key) {
-			nodes = append(nodes, c)
-		} else {
-			runtime.Log("irCandidateRemove: candidate has been removed")
-		}
+	prefix := []byte(candidatesKey)
+	stKey := append(prefix, key...)
+	if storage.Get(ctx, stKey) != nil {
+		storage.Delete(ctx, stKey)
+		runtime.Log("irCandidateRemove: candidate has been removed")
```

Comment:
This code should be executed after `common.RemoveVotes(ctx, hashID)`

For notary disabled environment, there is a pattern to prepare data before (to decrease extra GAS addition) `common.RemoveVotes(ctx, hashID)` but store in in the contract memory after. This is way we were preparing new list there, but actually update it (`common.SetSerialized()`) in the end.

Reply:
Fixed.